sound/soc/sof/loader.c

```diff
@@ -247,7 +247,7 @@ static int snd_sof_fw_ext_man_parse(struct snd_sof_dev *sdev,
 			elem_hdr->type, elem_hdr->size);
 
 		if (elem_hdr->size < sizeof(*elem_hdr) ||
-		    iptr + elem_hdr->size > fw->size) {
+		    iptr + elem_hdr->size > (uintptr_t)fw->data + fw->size) {
```

Comment:
in fact, if we really want to safely check the `elem_hdr->size` value, I think we also need to make sure it isn't overflowing / wrapping here. If `elem_hdr->size` is something like `~0` both tests will pass but it's obviously invalid. I think it would be better to compare `elem_hdr->size` against `remaining`

Reply:
good idea, changed